src/main/java/com/capstone/assessment_service/service/impl/AssessmentServiceImpl.java

```diff
@@ -0,0 +1,87 @@
+package com.capstone.assessment_service.service.impl;
+
+import com.capstone.assessment_service.dto.assessment.AssessmentRequestDto;
+import com.capstone.assessment_service.dto.assessment.AssessmentResponseDto;
+import com.capstone.assessment_service.exception.CapsuleExistsException;
+import com.capstone.assessment_service.exception.CapsuleNotFoundException;
+import com.capstone.assessment_service.mapper.AssessmentMapper;
+import com.capstone.assessment_service.messaging.CreateAssessmentOnMoodleProducerEvent;
+import com.capstone.assessment_service.model.AssessmentEntity;
+import com.capstone.assessment_service.model.SkillCapsuleEntity;
+import com.capstone.assessment_service.repository.AssessmentRepository;
+import com.capstone.assessment_service.service.AssessmentService;
+import com.capstone.assessment_service.service.CapsuleService;
+import lombok.RequiredArgsConstructor;
+import org.common.event.AssessmentEvent;
+import org.common.event.AssessmentUpdateEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class AssessmentServiceImpl implements AssessmentService {
+    private static final Logger logger = LoggerFactory.getLogger(AssessmentServiceImpl.class);
+    private final AssessmentRepository assessmentRepository;
+    private final CapsuleService capsuleService;
+    private final AssessmentMapper assessmentMapper;
+    private final CreateAssessmentOnMoodleProducerEvent createAssessmentOnMoodleProducerEvent;
+
+    @Override
+    public AssessmentResponseDto create(AssessmentRequestDto dto) {
+        if(findByName(dto.getAssessmentName()).isPresent()){
+            throw new CapsuleExistsException(
+                    String.format("An assessment with the name '%s' already exist",
+                            dto.getAssessmentName()));
+        }
```

Comment:
I understand the reasoning and necessity behind this logic, but you could leverage a database-level uniqueness constraint on the `name` field instead. By doing so, you’d only need to catch the conflict error and rethrow an appropriate exception. This approach would reduce unnecessary queries, optimize performance, and ensure the database enforces uniqueness where it matters most. Think of it as killing two birds :bird: with one stone. How cool is that?!! :smile: